timemachine/fe/free_energy.py

```diff
@@ -74,15 +74,21 @@ class PairBarPlots:
     overlap_detail_png: bytes
 
 
+@dataclass
+class IntermediateResult:
+    initial_states: List[InitialState]
+    pair_bar_results: List[PairBarResult]
+
+
 @dataclass
 class SimulationResult:
     initial_states: List[InitialState]
-    result: PairBarResult
+    pair_bar_results: List[PairBarResult]
```

Comment:
This change to make `PairBarResult` represent the result for a single pair of states simplifies the implementation of caching. If we want to avoid downstream changes, we could introduce a new internal type and avoid modifying this.